components/org.wso2.carbon.identity.media.core/src/main/java/org/wso2/carbon/identity/media/core/StorageSystemManager.java

```diff
@@ -274,17 +285,109 @@ public void validateFileUploadMediaTypes(String mediaTypePathParam, String conte
      */
     public void validateMediaTypePathParam(String mediaTypePathParam) throws StorageSystemClientException {
 
-        HashMap<String, List<String>> allowedContentTypes = StorageSystemUtil.getContentTypes();
-        if (MapUtils.isEmpty(allowedContentTypes) || !allowedContentTypes.keySet().contains(mediaTypePathParam)) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Started validating if the media content type path parameter in the request is a " +
+                    "supported content type.");
+        }
+
+        String envForAllowedContentTypes = System.getenv(CONFIGURABLE_MEDIA_CONTENT_TYPES);
+        Set<String> allowedContentTypes;
+        if (StringUtils.isNotBlank(envForAllowedContentTypes)) {
+            String[] contentTypes = envForAllowedContentTypes.split(",");
+            allowedContentTypes = new HashSet<>(Arrays.asList(contentTypes));
+
+        } else {
+            allowedContentTypes = StorageSystemUtil.getContentTypes().keySet();
+        }
+
+        if (CollectionUtils.isEmpty(allowedContentTypes) || !allowedContentTypes.contains(mediaTypePathParam)) {
             throw new StorageSystemClientException(String.format("Unsupported file content type: %s available as a " +
                     "path parameter in the request.", mediaTypePathParam));
         }
     }
 
-    private StorageSystemFactory getStorageSystemFactory(String storageType) {
+    /**
+     * Retrieve the username of the user who is making the request to the media service.
+     *
+     * @param username The username.
+     * @return user id.
+     * @throws StorageSystemServerException The server exception related retrieving user ID from username.
+     */
+    public String getUserIdFromUserName(String username) throws StorageSystemServerException {
 
-        return MediaServiceDataHolder.getInstance().getStorageSystemFactories().get(storageType);
+        try {
+            RealmService realmService = MediaServiceDataHolder.getInstance().getRealmService();
+            int tenantID = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+            UserRealm userRealm = realmService.getTenantUserRealm(tenantID);
+            if (userRealm != null) {
+                UserStoreManager userStoreManager = (UserStoreManager) userRealm.getUserStoreManager();
+                String userIdFromUserName = ((AbstractUserStoreManager) userStoreManager)
```

Comment:
does this work correctly for a secondary userstore user?

Reply:
yes I have tested the flow for a secondary userstore user